generator/mini_roundabout_processor.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "generator/feature_builder.hpp"
+#include "generator/mini_roundabout_info.hpp"
+#include "generator/osm_element.hpp"
+
+#include "coding/point_coding.hpp"
+
+#include "geometry/latlon.hpp"
+#include "geometry/mercator.hpp"
+#include "geometry/point2d.hpp"
+
+#include "base/math.hpp"
+
+#include <string>
+#include <vector>
+
+namespace generator
+{
+class MiniRoundaboutProcessor
+{
+public:
+  MiniRoundaboutProcessor();
+  MiniRoundaboutProcessor(double radiusMeters);
+
+  /// \brief Loads info about mini_roundabouts from binary source.
+  void ReadData(std::string const & intermediateFilePath);
```

Comment:
Эту функцию сделай плиз приватной, а конструктор пусть ее вызовет, кажется что без этого вызова ReadData с объектом MiniRoundaboutProcessor нет смысла работать

Reply:
ReadData может кидать исключения (в вызываются методы, по цепочке вызовов которых есть MYTHROW). Поэтому ее не надо добавлять в конструктор.

Reply:
```
Throwing an exception is the best way to signal an error during object construction. (Since there's no return value, there's no other way, other than constructing a headless object, which is bad style in C++.)

From the man himself, Bjarne Stroustrup: http://www.stroustrup.com/bs_faq2.html#ctor-exceptions
```
деструкторы другое дело

Reply:
Ты прав.